### DISCUSSION TITLE
Screenshot Macro

### DISCUSSION THREAD

**Discussion by Hlew7** [nickcoutsos/keymap-editor#249] [Q&A]
I'm a little confused with how to properly set up macros. I have read the zmk documentation multiple times, but the example does not make total sense to me.

I want to be able to long press one button (L Alt) and it press the combo "WinGUI+LShift+S" which pulls up windows screenshot. How would I do this with the configurator? Why is what I created not working?

Thanks
![Screenshot 2024-09-01 230455](https://github.com/user-attachments/assets/89e84516-4540-4192-9709-20dca6175a9b)

**Comment by nickcoutsos** [comment]
`&macro_tap` tells is to press each following key in sequence, ie: 

- press and release <kbd>WIN</kbd>, then
- press and release <kbd>SHIFT</kbd>, then
- press and release <kbd>S</kbd>

If you want them pressed together you can either include them in the `&macro_press` with `LALT` or (maybe more convenient) have a single `S` keypress that applies the <kbd>WIN</kbd> and <kbd>SHIFT</kbd> modifiers.

<img width="962" alt="Screen Shot 2024-09-02 at 10 27 29 AM" src="https://github.com/user-attachments/assets/67e4607b-daeb-4696-9a6f-05284db6d266">

Also, I"m not familiar with the Windows screenshot shortcuts, is <kbd>ALT</kbd> part of it or do you mean you want to long-press ALT to trigger this instead? If that's the case you'll want to implement this as a hold-tap behavior. You can actually get away with the default mod-tap implementation so there's no need to define a new behavior or macro:

<img width="663" alt="Screen Shot 2024-09-02 at 10 31 30 AM" src="https://github.com/user-attachments/assets/8baaafe3-195f-4f90-a9ab-fd0cc6d1934e">

**Reply by Hlew7** [reply]
Ah got it. I think in this case the Mod/Tap would be the way to go. ALT is not part of the key combo. I want to use a long-press ALT to trigger the shortcut, which is the combination of Shift+WIN+S. That being said, the single keypress with modifiers did the trick! Thanks!

Just so I understand, a macro is used to trigger a sequence of keys, whereas a Mod/Tap is used to trigger a combination of keys, correct? For example, I want to tap or press the right parentheses key (either/or), but when I do, I want both ( ) to appear, and I want the cursor to move to the middle of the bracket. Would that be a macro, or a mod/tap? What would that macro look like in the editor?

**Reply by nickcoutsos** [reply]
Macros are generally for sequences of keys, but they're very flexible so you can do things like combine and alternate between sequential and simultaneous key presses (or any behavior really), and even accept parameters.

In your example, it would seem like a simple sequence doing key presses for the open and close parentheses followed by one for the left arrow key.

<img width="572" alt="Screen Shot 2024-09-02 at 2 14 03 PM" src="https://github.com/user-attachments/assets/51c0bd5d-07c0-4223-a9e4-7d6a31085676">

Something like this I prefer to leave in the hands of my text editor as it can use context to see if injecting an extra character is appropriate, but this works as a simple macro example and I know plenty of people like using it.
